app/src/main/java/com/canvearth/canvearth/server/FBPixelManager.java

```diff
@@ -101,7 +118,58 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
         return fbPixel;
     }
 
-    public boolean writePixel(PixelData pixelData, Color color, @Nullable Runnable callback) {
+    // You don't have to watch this pixel (for now).. I'm nervous about performance issue of this method.
+    // returns Bitmap which has resolution of 2^resolutionFactor * 2^resolutionFactor
+    // TODO cache this when there is performance issue
+    // Do we need Async version of this?
+    public Bitmap getBitmapSync(PixelData pixelData, int resolutionFactor) {
+        int resolution = MathUtils.intPow(2, resolutionFactor);
+        final Bitmap bitmap = Bitmap.createBitmap(resolution, resolution, Bitmap.Config.ARGB_8888);
+        try {
+            int hierarchy = Math.min(Constants.LEAF_PIXEL_ZOOM_LEVEL - pixelData.zoom, resolutionFactor);
+            int chargeForOnePixel = MathUtils.intPow(2, resolutionFactor - hierarchy);
+            ArrayList<PixelData> childrenPixelData = PixelUtils.getChildrenPixelData(pixelData, hierarchy);
+            int childrenStartX = childrenPixelData.get(0).x;
+            int childrenStartY = childrenPixelData.get(0).y;
+            CountDownLatch latchForFinish = new CountDownLatch(childrenPixelData.size());
+            for (PixelData childPixelData: childrenPixelData) {
+                String firebaseId = childPixelData.firebaseId;
+                DatabaseUtils.getPixelReference(firebaseId).addListenerForSingleValueEvent(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                        Log.v(TAG, "Got one pixel");
+                        FBPixel ServerPixel4Firebase = dataSnapshot.getValue(FBPixel.class);
```

Comment:
Could you fix the inconsistencies in variable names?

Reply:
Got it.